src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -102,14 +108,18 @@ private function getIsGrantedString(IsGranted $isGranted): string
     {
         $processValue = fn ($value) => \sprintf($value instanceof Expression ? 'new Expression("%s")' : '"%s"', $value);
 
-        $argsString = $processValue($isGranted->attribute);
+        $argsString = $isGranted->callable ? 'callable' : $processValue($isGranted->attribute);
 
         if (null !== $subject = $isGranted->subject) {
-            $subject = !\is_array($subject) ? $processValue($subject) : array_map(function ($key, $value) use ($processValue) {
-                $value = $processValue($value);
-
-                return \is_string($key) ? \sprintf('"%s" => %s', $key, $value) : $value;
-            }, array_keys($subject), $subject);
+            if (\is_callable($subject)) {
```

Comment:
That's what I did at first. But if it comes to have a subject called `mail` [as pointed here](https://github.com/symfony/symfony/pull/59150#discussion_r1886644960), it would not work. Using `instanceof Closure` works well and avoids this problem.

Reply:
@alexandre-daubois that's precisely why @mtarld asks you to change your code to **use** `instanceof Closure` to avoid problems.

Reply:
Oh yes, I forgot to commit this one, that's why I read the other way around. 🤦 I'll do the update, sorry!